internal/provider/schema_cloudflare_tunnel_config.go

```diff
@@ -0,0 +1,177 @@
+									Default:     "",
+								},
+								"ca_pool": {
+									Type:        schema.TypeString,
+									Optional:    true,
+									Description: "Path to the certificate authority (CA) for the certificate of your origin. This option should be used only if your certificate is not signed by Cloudflare.",
+									Default:     "",
+								},
+								"no_tls_verify": {
+									Type:        schema.TypeBool,
+									Optional:    true,
+									Description: "Disables TLS verification of the certificate presented by your origin. Will allow any certificate from the origin to be accepted.",
+									Default:     false,
+								},
+								"disable_chunked_encoding": {
+									Type:        schema.TypeBool,
+									Optional:    true,
+									Description: "Disables chunked transfer encoding. Useful if you are running a Web Server Gateway Interface (WSGI) server.",
+									Default:     false,
+								},
+								"bastion_mode": {
+									Type:        schema.TypeBool,
+									Optional:    true,
+									Description: "Runs as jump host",
+								},
+								"proxy_address": {
+									Type:        schema.TypeString,
+									Optional:    true,
+									Description: "cloudflared starts a proxy server to translate HTTP traffic into TCP when proxying, for example, SSH or RDP. This configures the listen address for that proxy.",
+									Default:     "127.0.0.1",
+								},
+								"proxy_port": {
+									Type:        schema.TypeInt,
+									Optional:    true,
+									Description: "cloudflared starts a proxy server to translate HTTP traffic into TCP when proxying, for example, SSH or RDP. This configures the listen port for that proxy. If set to zero, an unused port will randomly be chosen.",
+									Default:     0,
+								},
+								"proxy_type": {
+									Type:         schema.TypeString,
+									Optional:     true,
+									Description:  "cloudflared starts a proxy server to translate HTTP traffic into TCP when proxying, for example, SSH or RDP. This configures what type of proxy will be started. Valid options are:.",
+									ValidateFunc: validation.StringInSlice([]string{"", "socks"}, false),
+									Default:      "",
+								},
+								"ip_rules": {
+									Type:        schema.TypeList,
+									Optional:    true,
+									Description: "IP rules for the proxy service",
+									Elem: &schema.Resource{
+										Schema: map[string]*schema.Schema{
+											"prefix": {
+												Type:     schema.TypeString,
+												Optional: true,
+											},
+											"ports": {
+												Type:     schema.TypeList,
+												Optional: true,
+												Elem:     &schema.Schema{Type: schema.TypeInt},
+											},
+											"allow": {
+												Type:     schema.TypeBool,
+												Optional: true,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+					"ingress_rule": {
+						Type:        schema.TypeList,
```

Comment:
if we don't care for ordering, let's use `TypeSet` here instead.

Reply:
Ordering does matter for `ingress_rule` as the config requires a catch-all rule, which if first, caused the API to return an error code.

Reply:
Docs: https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/install-and-setup/tunnel-guide/local/local-management/ingress/#matching-traffic